### DISCUSSION TITLE
EMMC access SD Card on RPi4 64 bit.

### DISCUSSION THREAD

**Discussion by wzydhek** [rsta2/circle#395] [Q&A] [answered]
I am trying to use the addon "SDCard" EMMC device to access the sdcard in my RPi4. I do NOT have USE_SDHOST eanbled as the SDCard addon won't compile with that option turned on.

In my kernel, all I am doing so far is Initializing the CEMMCDevice. Initialization fails:

logger: Circle 45.2 started on Raspberry Pi 4 Model B (AArch64)
00:00:00.66 timer: SpeedFactor is 1.51
00:00:00.67 emmc: Vendor 10, SD version 2, slot status 0
00:00:00.67 emmc: Resetting controller
00:00:00.67 emmc: control0: 00800000, control1: 00000000, control2: 00000000
00:00:00.68 emmc: checking for an inserted card
00:00:00.69 emmc: status: 1fff0000
00:00:00.69 emmc: Base clock rate is 100000000 Hz
00:00:00.70 emmc: setting clock rate
00:00:00.70 emmc: base_clock: 100000000, target_rate: 400000, divisor: 00000080, actual_clock: 390625, ret: 00008000
00:00:00.71 emmc: control0: 00800f00, control1: 000b8003
00:00:00.71 emmc: enabling SD clock
00:00:00.72 emmc: card insertion detected
00:00:00.73 emmc: Issuing command CMD0
00:00:00.73 emmc: command completed successfully
00:00:00.73 emmc: sending inquiry ACMD41
00:00:00.74 emmc: Issuing command CMD1
00:00:00.74 emmc: Error occured whilst waiting for command complete interrupt
00:00:00.75 emmc: Error issuing CMD1 (intr 00018000)
00:00:00.76 emmc: CMD_TIMEOUT
00:00:00.76 emmc: Inquiry ACMD41 failed

Not sure what I am doing wrong.

**Comment by wzydhek** [comment]
Appears to have the same problem when I compiled it for and ran it on my RPi3b+

logger: Circle 45.2 started on Raspberry Pi 3 Model B+ (AArch64)
00:00:00.66 timer: SpeedFactor is 1.51
00:00:00.66 emmc: Vendor 99, SD version 2, slot status 0
00:00:00.67 emmc: Resetting controller
00:00:00.67 emmc: control0: 00000000, control1: 00000000, control2: 00000000
00:00:00.68 emmc: checking for an inserted card
00:00:00.69 emmc: status: 1fff0000
00:00:00.69 emmc: Base clock rate is 200000000 Hz
00:00:00.69 emmc: setting clock rate
00:00:00.70 emmc: base_clock: 200000000, target_rate: 400000, divisor: 00000100, actual_clock: 390625, ret: 00000040
00:00:00.71 emmc: control0: 00000000, control1: 000b0043
00:00:00.71 emmc: enabling SD clock
00:00:00.72 emmc: card insertion detected
00:00:00.73 emmc: Issuing command CMD0
00:00:00.73 emmc: command completed successfully
00:00:00.73 emmc: sending inquiry ACMD41
00:00:00.74 emmc: Issuing command CMD1
00:00:00.74 emmc: Error occured whilst waiting for command complete interrupt
00:00:00.75 emmc: Error issuing CMD1 (intr 00018000)
00:00:00.76 emmc: CMD_TIMEOUT
00:00:00.76 emmc: Inquiry ACMD41 failed

**Comment by wzydhek** [comment, marked as answer]
I found that USE_EMBEDDED_MMC_CM was enabled. Disabled that, and it seems to work fine on both RPi3b+ and RPi4b

**Comment by rsta2** [comment]
For info: `USE_EMBEDDED_MMC_CM` is meant to be used to access the eMMC memory of Compute Module 3+ and 4. This does not work with the SD card. `USE_SDHOST` is not supported for Raspberry Pi 4, but is not necessary, because this model has a dedicated EMMC controller for the SD card.
